coretex/cli/modules/utils.py

```diff
@@ -17,20 +17,50 @@
 
 from typing import List, Any, Optional, Callable
 from functools import wraps
+from importlib.metadata import version as getLibraryVersion
 
 import sys
 
 from py3nvml import py3nvml
 
 import click
+import requests
 
+from . import ui
 from ...utils.process import command
 
 
 def updateLib() -> None:
     command([sys.executable, "-m", "pip", "install", "--no-cache-dir", "--upgrade", "coretex"], ignoreStdout = True, ignoreStderr = True)
 
 
+def fetchLatestVersion() -> Optional[str]:
+    url = "https://pypi.org/pypi/coretex/json"
+    response = requests.get(url)
+
+    if response.ok:
+        data = response.json()
+        return str(data.get("info", None).get("version", None))
+    else:
+        return None
+
+
+def checkLibVersion() -> None:
+    current = getLibraryVersion("coretex")
+    latest = fetchLatestVersion()
+
+    if latest is None:
+        return
+
+    majorCurrent, minorCurrent, patchCurrent = map(int, current.split('.'))
+    majorLatest, minorLatest, patchLatest = map(int, latest.split('.'))
```

Comment:
Validate both `current` and `latest` strings before splitting and converting them to integers

Reply:
Function getLibraryVersion() is returning "string" by default, when I tried to validate this variable I got mypy error that block of code in if statement is unreachable, is there really need to validate "current" variable?

Reply:
There is a need because what if versions has only 2 components, not 3 as we are expecting in our code? What if one of the version components cannot be converted to int?